runners/flink/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java

```diff
@@ -55,7 +55,13 @@ public static TestFlinkRunner create(boolean streaming) {
   @Override
   public PipelineResult run(Pipeline pipeline) {
     try {
-      return delegate.run(pipeline);
+      PipelineResult job = delegate.run(pipeline);
+
+      job.waitUntilFinish();
+//      job.waitUntilFinish(
```

Comment:
Remove this or even better create a second test for the with Duration signature.
A test for timeUntilFinish with timeOut like the one on direct runner seems also a good idea to have.
See https://github.com/apache/beam/pull/2408/files